pkg/sources/source_manager.go

```diff
@@ -207,6 +215,16 @@ func (s *SourceManager) ScanChunk(chunk *Chunk) {
 	s.outputChunks <- chunk
 }
 
+// AvailableCapacity returns the number of concurrent jobs the manager can
+// accommodate at this time. If there is no limit, -1 is returned.
+func (s *SourceManager) AvailableCapacity() int {
```

Comment:
suggestion: Could we avoid in-band errors here and return multiple values?
maybe:
```
func (s *SourceManager) AvailableCapacity() (int, bool) {
	if s.poolLimit == 0 {
		return 0, false
	}
	runCount := atomic.LoadInt32(&s.currentRunningCount)
	return s.poolLimit - int(runCount), true
}
```
ref: https://google.github.io/styleguide/go/decisions#in-band-errors